lib/Parse/ParseExpr.cpp

```diff
@@ -3324,9 +3283,19 @@ ParserResult<Expr> Parser::parseExprCollection(SourceLoc LSquareLoc) {
                DictionaryExpr::create(Context, LSquareLoc, {}, {}, RSquareLoc));
   }
 
+  // [#identifier is likely to be a legacy object literal.
```

Comment:
It's not guaranteed to, though, right? Do we still allow `[#selector(NSObject.copy)]`? Or was that never allowed in the first place?

Reply:
`[#selector` is tokenized as `l_square, pound_selector` in `Lexer`.

Reply:
I will add test cases for them anyway.